### PR TITLE
Port HttpClient v0 - v4 to F#

### PR DESCRIPTION
**Pull request by StachuDotNet**
Relates to #3242. WIP.
The legacy HttpClient versions (v0 - v4) need to be ported to F# for the great F# migration.

### CODE REVIEW

**Comment by StachuDotNet** [comment]
@pbiggar I'm seeing a number of HttpClient fns (both v5 and legacy) whose `sqlSpec`s are `NotYetImplementedTODO`. I would expect them to be `NotQueryable`. Should I go ahead and update to `NotQueryable` or am I mistaken?

**Comment by pbiggar** [comment]
> @pbiggar I'm seeing a number of HttpClient fns (both v5 and legacy) whose `sqlSpec`s are `NotYetImplementedTODO`. I would expect them to be `NotQueryable`. Should I go ahead and update to `NotQueryable` or am I mistaken?

Yes, you can make them NotQueryable, thanks!

File: fsharp-backend/src/BackendOnlyStdLib/LibHttpClient5.fs [hunk 1]
```diff
@@ -22,7 +22,7 @@ let err (str : string) = Ply(Dval.errStr str)
 let incorrectArgs = Errors.incorrectArgs
 
 let varA = TVariable "a"
-let varB = TVariable "b"
```
Comment (StachuDotNet) [inline fsharp-backend/src/BackendOnlyStdLib/LibHttpClient5.fs:25]: this was wholly unused

**Comment by StachuDotNet** [comment]
This branch has now been caught up with latest on `main`, and commit history is a bit cleaner.

At this point, there are some remaining TODOs:
- there are a number of `_`-disabled tests in the various `httpclienttestfiles/vX` folder
- tests involving fuzzers should be added, to find and resolve any other outstanding differences between the OCaml and F# implementations

**Comment by StachuDotNet** [comment]
Additionally, it seems that CI won't run for my branch. Is that expected @pbiggar?

**Comment by pbiggar** [comment]
I don't remember what circle does with forks but I'll take a look. Obviously we should make it run

File: fsharp-backend/src/BackendOnlyStdLib/LegacyBaseHttpClient.fs [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+    HttpHeaders.get "content-type" headers
+    |> Option.map (fun s -> s = "application/x-www-form-urlencoded")
+    |> Option.defaultValue false
+
+
+
+// includes an implicit content-type
+type Content =
+  // OCaml's impl. uses cURL under the hood.
+  // cURL is special_ in that it will assume that
```
Comment (StachuDotNet) [inline fsharp-backend/src/BackendOnlyStdLib/LegacyBaseHttpClient.fs:134]:
```suggestion
  // cURL is special in that it will assume that
```

**Pull request merged**